dask/dataframe/optimize.py

```diff
@@ -19,7 +20,11 @@
         # Merge column access into bcolz loading
         RewriteRule((getitem, (dataframe_from_ctable, a, b, c, d), e),
                     (dataframe_from_ctable, a, b, e, d),
-                    (a, b, c, d, e)))
+                    (a, b, c, d, e)),
+        RewriteRule((getitem, (Castra.load_partition, '~c', '~part', '~cols1'),
```

Comment:
Yes, I plan to rip out `pframe` and replace it with a combination of `partd` and `castra`.

Sorry for the on-disk data structure churn.  I don't expect these to be the last iterations.